slack-plugin/src/main/kotlin/slack/gradle/tasks/BaseDependencyCheckTask.kt

```diff
@@ -15,53 +15,47 @@
  */
 package slack.gradle.tasks
 
+import com.android.build.gradle.internal.publishing.AndroidArtifacts
+import com.android.build.gradle.internal.publishing.AndroidArtifacts.ArtifactType
 import org.gradle.api.DefaultTask
 import org.gradle.api.artifacts.Configuration
-import org.gradle.api.artifacts.component.ModuleComponentIdentifier
-import org.gradle.api.tasks.Internal
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.TaskAction
-import org.gradle.maven.MavenModule
+import org.gradle.internal.component.external.model.ModuleComponentArtifactIdentifier
 
 public abstract class BaseDependencyCheckTask : DefaultTask() {
-  @get:Internal public lateinit var configuration: Configuration
-
-  internal abstract fun handleDependencies(dependencies: Map<String, String>)
-
-  init {
-    @Suppress("LeakingThis")
-    notCompatibleWithConfigurationCache(
-      "Pending Gradle 7.5 and https://github.com/gradle/gradle/pull/18729"
+  @get:Input public abstract val identifiersToVersions: MapProperty<String, String>
+
+  protected abstract fun handleDependencies(identifiersToVersions: Map<String, String>)
+
+  protected fun configureIdentifiersToVersions(configuration: Configuration) {
+    identifiersToVersions.putAll(
+      configuration.incoming
+        .artifactView {
+          attributes { attribute(AndroidArtifacts.ARTIFACT_TYPE, ArtifactType.AAR_OR_JAR.type) }
+          lenient(true)
+        }
```

Comment:
It looks like you might only care about external dependencies? (based on the filter at line 47) In that case, consider this:
```kotlin
// Only resolve external dependencies! Without this, all project dependencies will get _compiled_.
view.componentFilter { id -> id is ModuleComponentIdentifier }
```
That's a code snippet from an internal plugin I maintain for resolving dependencies.

Reply:
oh neat, thanks

Reply:
Does this look right? [ef0ad80](https://github.com/slackhq/slack-gradle-plugin/pull/23/commits/ef0ad804c62767660311d612de744c173d82cb90)